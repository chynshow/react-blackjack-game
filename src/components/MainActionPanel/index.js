import React, { useContext } from 'react';
import { AppContext } from '../../state/AppContext';

const MainActionPanel = () => {
  const {
    startGame,
    resetGame,
    saveGame,
    loadGame,
    showGameScore,
    state: { gameStarted, gameSave, gameScore },
  } = useContext(AppContext);

  return (
    <div className='c-main-action-panel'>
      {!gameStarted && (
        <button
          className='c-btn c-main-action-panel__btn'
          onClick={() => startGame()}
        >
          Start Game
        </button>
      )}
      {gameStarted && (
        <>
          <button
            className='c-btn c-main-action-panel__btn'
            disabled={!gameStarted}
            onClick={() => resetGame()}
          >
            Reset Game
          </button>
          {!gameScore.length <= 0 && (
            <button
              className='c-btn c-main-action-panel__btn'
              disabled={gameScore.length <= 0}
              onClick={() => showGameScore()}
            >
              Score
            </button>
          )}
          <button
            className='c-btn c-main-action-panel__btn'
            onClick={() => saveGame()}
          >
            Save Game
          </button>
          {gameSave && (
            <button
              className='c-btn c-main-action-panel__btn'
              onClick={() => loadGame()}
            >
              Load Game
            </button>
          )}
        </>
      )}
    </div>
  );
};

export default MainActionPanel;
