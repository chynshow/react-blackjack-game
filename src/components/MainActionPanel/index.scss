.c-main-action-panel {
  padding: 1rem 0;

  &__start-btn {
    animation: fromRight 1000ms linear both;
  }
}
