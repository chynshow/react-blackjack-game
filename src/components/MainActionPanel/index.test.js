import React from 'react';
import MainActionPanel from './index';
import { shallow } from 'enzyme';

it('render MainActionPanel component', () => {
  const c = shallow(<MainActionPanel />);
  expect(c).toMatchSnapshot();
});
