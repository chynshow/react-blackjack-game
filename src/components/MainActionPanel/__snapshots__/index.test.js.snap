// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainActionPanel component 1`] = `
<div
  className="c-main-action-panel"
>
  <button
    className="c-btn c-main-action-panel__start-btn"
    disabled={false}
    onClick={[Function]}
  >
    Start Game
  </button>
</div>
`;
